### PR TITLE
Use LOG_ constants infavour of PRINTF for rpl-classic

### PR DESCRIPTION
**Pull request by lafka**
Debugging could only be enabled pr. file at compile time. Refactor to use sys/log.h module. Also removes some regular printfs.

Some questions though:

* Replacing `PRINTLLADDR((uip_lladdr_t *) _)` with `LOG_DBG_LLADDR(_)`, i've seen it in rpl-lite but i'm wondering if this is correct?
* Where variables are used exclusively withing a `LOG_DBG_ENABLED` block should the definition be within the `#if` block or should i add another `#if` at the top of function with the variable declaration (see https://github.com/contiki-ng/contiki-ng/compare/develop...lafka:lafka-debug-fix?expand=1#diff-98a2a62e3c8b2de1cc3b9efe231c97fbR67)
* `handle_periodic_timer` in `rpl-nbr-policy.c`, should this really be there? it changes functionality based on LOG_LEVEL

### CODE REVIEW

**Comment by simonduq** [comment]
Thanks!
Comments:
* Please remove the prefixes in log strings, as the log module will already prepend `RPL`
* Please use different log levels for different event types, not just `DBG` everywhere

As for your questions:
1. yes
1. declaring the var directly in the same block works for me
1. not sure (haven't looked in detail)

File: os/net/routing/rpl-classic/rpl-timers.c [hunk 1]
```diff
@@ -180,23 +181,23 @@ handle_dio_timer(void *ptr)
...
     if(instance->dio_intcurrent < instance->dio_intmin + instance->dio_intdoubl) {
       instance->dio_intcurrent++;
-      PRINTF("RPL: DIO Timer interval doubled %d\n", instance->dio_intcurrent);
+      LOG_DBG("DIO Timer interval doubled %d\n", instance->dio_intcurrent);
     }
     new_dio_interval(instance);
   }
 
-#if DEBUG
+#if LOG_DBG_ENABLED
```
Comment (lafka) [inline os/net/routing/rpl-classic/rpl-timers.c]: rpl-lite uses `if (LOG_DBG_ENABLED) {/* .. */ }`. I prefer the use of preprocessor as it discards the entire body and makes it easier for optimizer to eliminate dead code. That being said, I'm not sure exactly what the optimizer does when encountering a `if` with a constant condition.

**Comment by lafka** [comment]
Updated

**Comment by simonduq** [comment]
OK, one last thing that will need fixing: `LOG_*_ENABLED` should be checked in `if()` statement, not pre-processor `#if`. This is because of how log levels are handled at runtime (e.g. can be changed via the shell).

**Comment by lafka** [comment]
Updated

File: os/net/routing/rpl-classic/rpl-dag-root.c [hunk 2]
```diff
@@ -64,13 +64,17 @@ set_global_address(uip_ipaddr_t *prefix, uip_ipaddr_t *iid)
...
   uip_ds6_addr_add(&root_ipaddr, 0, ADDR_AUTOCONF);
 
-  printf("IPv6 addresses: ");
-  for(i = 0; i < UIP_DS6_ADDR_NB; i++) {
-    state = uip_ds6_if.addr_list[i].state;
-    if(uip_ds6_if.addr_list[i].isused &&
-       (state == ADDR_TENTATIVE || state == ADDR_PREFERRED)) {
-      uip_debug_ipaddr_print(&uip_ds6_if.addr_list[i].ipaddr);
-      printf("\n");
+  if (LOG_DBG_ENABLED) {
```
Comment (simonduq) [inline os/net/routing/rpl-classic/rpl-dag-root.c]: code style (here and other locations in this PR): no space between `if` and `(`

File: os/net/routing/rpl-classic/rpl-dag.c [hunk 3]
```diff
@@ -543,17 +542,17 @@ int
 rpl_set_default_route(rpl_instance_t *instance, uip_ipaddr_t *from)
 {
   if(instance->def_route != NULL) {
-    PRINTF("RPL: Removing default route through ");
-    PRINT6ADDR(&instance->def_route->ipaddr);
-    PRINTF("\n");
+    LOG_DBG_("Removing default route through ");
```
Comment (simonduq) [inline os/net/routing/rpl-classic/rpl-dag.c]: this should start a new line, i.e., no ending `_`

File: os/net/routing/rpl-classic/rpl-dag.c [hunk 4]
```diff
@@ -543,17 +542,17 @@ int
...
+    LOG_DBG_("\n");
     uip_ds6_defrt_rm(instance->def_route);
     instance->def_route = NULL;
   }
 
   if(from != NULL) {
-    PRINTF("RPL: Adding default route through ");
-    PRINT6ADDR(from);
-    PRINTF("\n");
+    LOG_DBG_("Adding default route through ");
```
Comment (simonduq) [inline os/net/routing/rpl-classic/rpl-dag.c]: same here, no ending `_`

File: os/net/routing/rpl-classic/rpl-dag.c [hunk 5]
```diff
@@ -661,9 +660,9 @@ void
 rpl_free_dag(rpl_dag_t *dag)
 {
   if(dag->joined) {
-    PRINTF("RPL: Leaving the DAG ");
-    PRINT6ADDR(&dag->dag_id);
-    PRINTF("\n");
+    LOG_INFO_("Leaving the DAG ");
```
Comment (simonduq) [inline os/net/routing/rpl-classic/rpl-dag.c]: no ending `_` here

File: os/net/routing/rpl-classic/rpl-dag.c [hunk 6]
```diff
@@ -792,9 +791,9 @@ rpl_select_dag(rpl_instance_t *instance, rpl_parent_t *p)
       rpl_remove_routes(instance->current_dag);
     }
 
-    PRINTF("RPL: New preferred DAG: ");
-    PRINT6ADDR(&best_dag->dag_id);
-    PRINTF("\n");
+    LOG_INFO_("New preferred DAG: ");
```
Comment (simonduq) [inline os/net/routing/rpl-classic/rpl-dag.c]: no ending `_` here

File: os/net/routing/rpl-classic/rpl-dag.c [hunk 7]
```diff
@@ -1564,9 +1563,9 @@ rpl_process_dio(uip_ipaddr_t *from, rpl_dio_t *dio)
 
   /* The DIO comes from a valid DAG, we can refresh its lifetime */
   dag->lifetime = (1UL << (instance->dio_intmin + instance->dio_intdoubl)) * RPL_DAG_LIFETIME / 1000;
-  PRINTF("Set dag ");
-  PRINT6ADDR(&dag->dag_id);
-  PRINTF(" lifetime to %ld\n", dag->lifetime);
+  LOG_INFO_("Set dag ");
```
Comment (simonduq) [inline os/net/routing/rpl-classic/rpl-dag.c]: no ending `_` here

**Comment by simonduq** [comment]
There are a couple misuses of the ending `_` in logs. Might be a good idea to run a simulation in Cooja with max log level, and check if the serial output looks sane.

**Comment by simonduq** [comment]
Also, Travis is not happy about something :(

File: os/net/routing/rpl-classic/rpl-dag.c (hunk 3 above)
Comment (lafka) [reply os/net/routing/rpl-classic/rpl-dag.c]: It immediately logs the ip6 address for the route and prints newline afterwards. Is there a different way i should be calling this to log the address?

Comment (simonduq) [reply os/net/routing/rpl-classic/rpl-dag.c]: no, talking about the line above the address logging. Should be
```
LOG_DBG("Removing default route through ");
```

Comment (lafka) [reply os/net/routing/rpl-classic/rpl-dag.c]: Oh my confusion! For some reason i though the `LOG_DBG()` automatically added newline. Fixing.

**Comment by lafka** [comment]
Logs are fixed and updated. I'm looking into the travis output, it seems to fail related a few variables that are undeclared or unused. Will replicate locally and fix it up!

**Pull request merged**